lib/rhc/ssh_helpers.rb

```diff
@@ -256,29 +257,43 @@ def save_snapshot(app, filename, for_deployment=false, ssh_executable=nil)
       ssh_stderr = " 2>/dev/null"
       debug ssh_cmd
 
+      # Default timeout is 30 minutes, more than enough time to ensure the snapshot is copied successfully.
+      options.timeout ? exec_timeout = options.timeout : exec_timeout = 1800
+
       say "Pulling down a snapshot of application '#{app.name}' to #{filename} ... "
 
       begin
         if !RHC::Helpers.windows?
+          status = 1
+          output = ""
+          Timeout::timeout(exec_timeout) {
             status, output = exec(ssh_cmd + (debug? ? '' : ssh_stderr))
-            if status != 0
-              debug output
-              raise RHC::SnapshotSaveException.new "Error in trying to save snapshot. You can try to save manually by running:\n#{ssh_cmd}"
-            end
+          }
+          if status != 0
+            debug output
+            raise RHC::SnapshotSaveException.new "Error in trying to save snapshot. You can try to save manually by running:\n#{ssh_cmd}"
+          end
         else
-          Net::SSH.start(ssh_uri.host, ssh_uri.user) do |ssh|
-            File.open(filename, 'wb') do |file|
-              ssh.exec! snapshot_cmd do |channel, stream, data|
-                if stream == :stdout
-                  file.write(data)
-                else
-                  debug data
+          Timeout::timeout(exec_timeout) {
```

Comment:
It's a minor thing, but it's generally preferable to use `do`/`end` instead of braces for long blocks like this one.